pulsar-common/src/main/java/org/apache/pulsar/common/util/FutureUtil.java

```diff
@@ -47,6 +50,18 @@ public static CompletableFuture<Void> waitForAll(Collection<? extends Completabl
         return CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
     }
 
+    public static <T> CompletableFuture<List<T>> waitForAll(Stream<CompletableFuture<List<T>>> futures) {
+        return futures.reduce(CompletableFuture.completedFuture(new ArrayList<>()),
+                (pre, curr) -> pre.thenCompose(preV -> curr.thenApply(currV -> {
+                    preV.addAll(currV);
+                    return preV;
+                })));
+    }
+
+    public static <T> CompletableFuture<List<T>> waitForAll(List<CompletableFuture<List<T>>> futures) {
+        return waitForAll(futures.stream());
```

Comment:
And here.

Reply:
Yes, I will fix it and mark it resolved.